### PR TITLE
Upstream verconf patch from grpc/grpc

### PR DESCRIPTION
**Pull request by qnighy**
This patch upstreams @jtattermusch's https://github.com/grpc/grpc/pull/12779.

### CODE REVIEW

File: tasks/bin/cross-ruby.rake [hunk 1]
```diff
@@ -130,6 +130,9 @@ end
 # make
 file "#{build_dir}/ruby.exe" => ["#{build_dir}/Makefile"] do |t|
   chdir File.dirname(t.prerequisites.first) do
+    # From https://github.com/grpc/grpc/issues/12161. Old rubies have race conditions around verconf.h.
```
Comment (kou) [inline tasks/bin/cross-ruby.rake:133]: Which versions are "Old rubies"?

Comment (qnighy) [reply tasks/bin/cross-ruby.rake]: I looked into it as I didn't know the details. It turned out the "old rubies" here are 2.3.x or before. Added a comment about it.

Comment (qnighy) [reply tasks/bin/cross-ruby.rake]: 2.3.0 is quite old to me, but protocolbuffers/protobuf still seems to provide builds for them https://github.com/protocolbuffers/protobuf/blob/v3.14.0/ruby/Rakefile#L84

Comment (kou) [reply tasks/bin/cross-ruby.rake]: OK.
We don't support EOE-ed Rubies.

Comment (qnighy) [reply tasks/bin/cross-ruby.rake]: I see. Thanks anyway for reviewing this!

**Pull request closed without merging**